frontend/src/scenes/funnels/funnelLogic.js

```diff
@@ -0,0 +1,72 @@
+import { kea } from 'kea'
+import api from 'lib/api'
+import { toParams } from 'lib/utils';
+
+
+
+export const funnelLogic = kea({
+    key: props => props.id,
+    actions: () => ({
+        setFilters: input => input
+    }),
+    loaders: ({ props, values }) => ({
+        funnel: {
+            loadFunnel: async (id = props.id) => {
+                return await api.get('api/funnel/' + id + '/?exclude_count=1')
+            },
+        },
+        steps: {
+            loadSteps: async (id = props.id) => {
+                return await api.get('api/funnel/' + id + '/?' + toParams(values.filters))
+            },
+        },
+        people: {
+            loadPeople: async (steps) => {
+                return await api.get('api/person/?id=' + steps[0].people.slice(0, 99).join(','))
+            }
+        },
+    }),
+    reducers: ({ actions }) => ({
+        filters: [
+            {},
+            {
+                [actions.setFilters]: (state, input) => {
+                    return { ...state, ...input }
+                },
+            }
+        ],
+        steps: [
+            null,
+            {
+                [actions.loadStepsSuccess]: (_, { steps }) => {
+                    return steps.steps
+                }
+            }
+        ],
+        people: [
+            null,
+            {
+                [actions.loadPeopleSuccess]: (_, { people }) => {
+                    console.log('loadingpeople', people)
+                    return people.results
+                }
+            }
+        ],
+    }),
+    listeners: ({ actions, values }) => ({
+        [actions.loadStepsSuccess]: async ({ steps }, breakpoint) => {
+            breakpoint();
+            await actions.loadPeople(values.steps)
```

Comment:
There's nothing to wait here. Calling an action just dispatches it... and then it's up to whatever listener to react to it. The mental model to have is more of dispatching events than running thunks that return a value.

No need for a breakpoint either in this case. Usually a breakpoint is something you put between two `await` calls to not call the second one in case the action (`loadStepsSuccess` in this case) got dispatched again